core/src/SYCL/Kokkos_SYCL_Team.hpp

```diff
@@ -182,6 +182,44 @@ class SYCLTeamMember {
     m_item.barrier(sycl::access::fence_space::local_space);
   }
 
+  // FIXME_SYCL move somewhere else and combine with other places that do
+  // parallel_scan
+  // Exclusive scan returning the total sum.
+  // n is required to be a power of two and
+  // temp must point to an array containing the data to be processed
+  // The accumulated value is returned.
+  template <typename Type>
+  static Type prescan(sycl::nd_item<2> m_item, Type* temp, int n) {
+    int thid = m_item.get_local_id(0);
+
+    // First do a reduction saving intermediate results
+    for (int stride = 1; stride < n; stride <<= 1) {
+      auto idx = 2 * stride * (thid + 1) - 1;
+      if (idx < n) temp[idx] += temp[idx - stride];
+      m_item.barrier(sycl::access::fence_space::local_space);
+    }
+
+    Type total_sum = temp[n - 1];
+    m_item.barrier(sycl::access::fence_space::local_space);
+
+    // clear the last element so we get an exclusive scan
+    if (thid == 0) temp[n - 1] = 0;
```

Comment:
Is "assignable from 0" part of the requirements on the type used in a team scan?

Reply:
I don't think we phrase any requirements clearly, yet. In view of https://github.com/kokkos/kokkos/pull/3602, it probably makes sense to provide an implementation based on a reducer with the default being `Sum`.

Reply:
Changed to brace initialization for now.